tokio/src/runtime/task/mod.rs

```diff
@@ -220,11 +366,13 @@ impl<S: Schedule> UnownedTask<S> {
     }
 
     pub(crate) fn run(self) {
+        let raw = self.raw;
+        mem::forget(self);
+
         // Decrement the ref-count
-        self.raw.header().state.ref_dec();
+        raw.header().state.ref_dec();
```

Comment:
We *probably* should decrement the ref-count *after* polling. Before, the ref-dec position didn't really matter in poll because we assumed the "task list" held the ref count, but I am not 100% sure if that still applies here.

Reply:
I did take care in the `poll` method that this should be fine, but I have swapped them to make the correctness more obvious.